### PR TITLE
refactor auto release to release only if all jobs were successful on master

### PR DESCRIPTION
**Pull request by flaviencathala**
refactor auto release to release only if all jobs were successful on master

### CODE REVIEW

**Comment by flaviencathala** [comment]
@jan-molak I have tweaked the NPM settings but nothing worked. It is very unlikely this refactoring will fix the release but I cannot spend more time on this today and at least the release will be done only once all tests have passed. Can you please have a quick look?

**Comment by flaviencathala** [comment]
@jan-molak I forgot to mention that I tried locally with the tokens from this repo's settings and the authentication worked well..

File: .github/workflows/build-and-release.yml [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+        uses: actions/setup-node@v2
+        with:
+          node-version: 14
+          registry-url: https://registry.npmjs.org/
+      - run: npm ci
+      - name: Release
+        run: npx semantic-release --debug
+        env:
+          CI: true
+          NPM_TOKEN: ${{ secrets.NPM_TOKEN }}
```
Comment (jan-molak) [inline .github/workflows/build-and-release.yml:44]: Hey, I just found [this](https://github.com/actions/setup-node/blob/main/action.yml#L16):

```
registry-url - Will set the registry in a project level .npmrc and .yarnrc file, 
               and set up auth to read in from env.NODE_AUTH_TOKEN
```

Should we change `NPM_TOKEN` to `NODE_AUTH_TOKEN` ?

**Review by jan-molak — CHANGES_REQUESTED** [review]

**Comment by jan-molak** [comment]
Hey @flaviencathala - I think the pipeline config might be using an incorrect name for the env variable:

From `setup-node` docs:
```
- run: npm publish
  env:
    NODE_AUTH_TOKEN: ${{ secrets.NPM_TOKEN }}  # note NODE_AUTH_TOKEN vs NPM_TOKEN
```

File: .github/workflows/build-and-release.yml (hunk 1 above)
Comment (flaviencathala) [reply .github/workflows/build-and-release.yml:44]: My understanding was that `semantic-release` was taking care of everything using `NPM_TOKEN`, and it did when I tried locally.
```
[semantic-release] [@semantic-release/npm] › ℹ  Verify authentication for registry https://registry.npmjs.org/
[semantic-release] [@semantic-release/npm] › ℹ  Wrote NPM_TOKEN to /private/var/folders/***/.npmrc
protractor-cucumber-framework-publisher
[semantic-release] › ✔  Completed step "verifyConditions" of plugin "@semantic-release/npm"
```
But at this point we can try to set both and see how it behaves.

**Comment by flaviencathala** [comment]
So apparently that was it, good catch @jan-molak!

**Comment by jan-molak** [comment]
Haha, we got there eventually, @flaviencathala 😄

**Pull request merged**